### PR TITLE
feat: cross chain staking V2, init commit

### PR DESCRIPTION
**Pull request by fb-alexcq**
Init commit. a few more updates coming.

### CODE REVIEW

File: contracts/WooStakingController.sol [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    event CompoundOnController(address indexed user);
+    event AdminUpdated(address indexed addr, bool flag);
+
+    modifier onlyAdmin() {
+        require(msg.sender == owner() || isAdmin[msg.sender], "WooStakingController: !admin");
+        _;
+    }
+
+    constructor(address _endpoint, address _rewardRouter) NonblockingLzApp(_endpoint) {
+        transferOwnership(msg.sender);
```
Comment (merlincdj) [inline contracts/WooStakingController.sol:38]: dont need this line, should remove.

File: contracts/WooStakingProxy.sol [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    event CompoundOnProxy(address indexed user);
+    event AdminUpdated(address indexed addr, bool flag);
+
+    modifier onlyAdmin() {
+        require(msg.sender == owner() || isAdmin[msg.sender], "WooStakingProxy: !admin");
+        _;
+    }
+
+    constructor(address _endpoint, uint16 _chainId, address _controller, address _want) NonblockingLzApp(_endpoint) {
+        transferOwnership(msg.sender);
```
Comment (merlincdj) [inline contracts/WooStakingProxy.sol:39]: dont need this line, should remove.

File: contracts/WooStakingProxy.sol [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    event WithdrawOnProxy(address indexed user, uint256 amount);
+    event CompoundOnProxy(address indexed user);
+    event AdminUpdated(address indexed addr, bool flag);
+
+    modifier onlyAdmin() {
+        require(msg.sender == owner() || isAdmin[msg.sender], "WooStakingProxy: !admin");
+        _;
+    }
+
+    constructor(address _endpoint, uint16 _chainId, address _controller, address _want) NonblockingLzApp(_endpoint) {
```
Comment (merlincdj) [inline contracts/WooStakingProxy.sol:38]: `_chainId` -> `_controllerChainId`

**Comment by merlincdj** [comment]
proxy need interface

File: contracts/WooStakingProxy.sol (hunk 3 above)
Comment (fb-alexcq) [reply contracts/WooStakingProxy.sol:38]: okay

File: contracts/WooStakingProxy.sol (hunk 2 above)
Comment (fb-alexcq) [reply contracts/WooStakingProxy.sol:39]: Okay. fixing.

**Pull request merged**